pilot/pkg/model/fuzz_test.go

```diff
@@ -35,17 +35,33 @@ func FuzzDeepCopyWorkloadInstance(f *testing.F) {
 }
 
 func FuzzDeepCopyIstioEndpoint(f *testing.F) {
-	fuzzDeepCopy[*IstioEndpoint](f)
+	fuzzDeepCopyEqual[*IstioEndpoint](f)
 }
 
 type deepCopier[T any] interface {
 	DeepCopy() T
 }
 
+type equalCopier[T any] interface {
+	deepCopier[T]
+	Equals(T) bool
+}
+
+// fuzzDeepCopy runs a fuzz test that, after calling DeepCopy, the result is equal
 func fuzzDeepCopy[T deepCopier[T]](f test.Fuzzer) {
 	fuzz.Fuzz(f, func(fg fuzz.Helper) {
 		orig := fuzz.Struct[T](fg)
 		copied := orig.DeepCopy()
 		assert.Equal(fg.T(), orig, copied)
 	})
 }
+
+// fuzzDeepCopyEqual runs a fuzz test that, after calling DeepCopy, the result is equal by both cmp.Equal and the struct-defined Equals() method
+func fuzzDeepCopyEqual[T equalCopier[T]](f test.Fuzzer) {
```

Comment:
Here we can only verify that the copy is correct, but not that it is a `deep' copy. I think the correct verification should be as follows:

```
orig := fuzz.Struct[T](fg)
fast := orig.DeepCopy()    // written by hand, very fast
slow := SlowDeepCopy(orig) // general deep copy util function(only used for test), it can guarantee a correct `deep' copy, but the performance may be poor


// verify the copy is correct, but is may be not `deep` copy
assert.Equal(fg.T(), orig, fast)
assert.Equal(fg.T(), orig, slow)


// verify it is a `deep` copy
_ = randModify(orig) // random modify util function
assert.Equal(fg.T(), fast, slow) // fast equals slow

```

Reply:
I submitted a PR for this https://github.com/istio/istio/pull/52535